src/modules/cpu/kernel/tensor_sum.hpp

```diff
@@ -0,0 +1,798 @@
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+#include "reduction.hpp"
+
+RppStatus tensor_sum_u8_u64_host(Rpp8u *srcPtr,
+                                 RpptDescPtr srcDescPtr,
+                                 Rpp64u *tensorSumArr,
+                                 RpptROIPtr roiTensorPtrSrc,
+                                 RpptRoiType roiType,
+                                 RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(srcDescPtr->n)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp8u *srcPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp8u *srcPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+
+        Rpp32u alignedLength = (bufferLength / 48) * 48;
+        Rpp32u vectorIncrement = 48;
+        Rpp32u vectorIncrementPerChannel = 16;
+
+        // Tensor Sum without fused output-layout toggle (NCHW)
+        if ((srcDescPtr->c == 1) && (srcDescPtr->layout == RpptLayout::NCHW))
+        {
+            alignedLength = bufferLength & ~15;
+            Rpp32u sum = 0;
+            Rpp32u sumAvx[8] = {0};
+
+            Rpp8u *srcPtrRow;
+            srcPtrRow = srcPtrChannel;
+#if __AVX2__
+            __m256i psum = _mm256_setzero_si256();
+#endif
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTemp;
+                srcPtrTemp = srcPtrRow;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+#if __AVX2__
+                    __m256i p1[2];
+                    rpp_simd_load(rpp_load16_u8_to_u32_avx, srcPtrTemp, p1);
+                    compute_sum_16_host(p1, &psum);
+#endif
+                    srcPtrTemp += vectorIncrementPerChannel;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    sum += (Rpp32u)(*srcPtrTemp++);
+                }
+                srcPtrRow += srcDescPtr->strides.hStride;
+            }
+#if __AVX2__
+            rpp_simd_store(rpp_store8_u32_to_u32_avx, sumAvx, &psum);
+#endif
+            sum += (sumAvx[0] + sumAvx[1] + sumAvx[2] + sumAvx[3] + sumAvx[4] + sumAvx[5] + sumAvx[6] + sumAvx[7]);
```

Comment:
sumAvx will be computed only for __AVX2__.  So this has to go inside the compile flag

Reply:
@rrawther This sum is just the very final addition - for the vector elements in the summed up AVX2 register.
The _mm256_hadd_ps() cannot be used since it only adds adjacent pairs of single-precision f32s. Its better to do the sum[0 through 7] at the end.